### PR TITLE
Modify the sierra bib & item merger to record to VHS 🎥

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Modify the hybrid item merger to run with VHS.

Part of #1473.

### Who is this change for?

🎥

### CODE REVIEW

File: common/docker-compose.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ elasticsearch:
     - "http.host=0.0.0.0"
     - "transport.host=0.0.0.0"
     - "cluster.name=wellcome"
+    - "ES_JAVA_OPTS=-Xms750m -Xmx750m"
```
Comment (kenoir) [inline common/docker-compose.yml]: This doesn't need to be here, it stops the common tests failing so often for me.

File: sierra_adapter/common/src/main/scala/uk/ac/wellcome/sierra_adapter/dynamo/SierraTransformableDao.scala [hunk 2]
```diff
@@ -1,67 +0,0 @@
-package uk.ac.wellcome.sierra_adapter.dynamo
```
Comment (kenoir) [inline sierra_adapter/common/src/main/scala/uk/ac/wellcome/sierra_adapter/dynamo/SierraTransformableDao.scala]: Bye Bye DAO

File: sierra_adapter/sierra_bib_merger/src/test/scala/uk/ac/wellcome/platform/sierra_bib_merger/SierraBibMergerFeatureTest.scala [hunk 3]
```diff
@@ -149,13 +191,20 @@ class SierraBibMergerFeatureTest
...
+    val expectedSierraTransformable =
       SierraTransformable(bibRecord = record, version = 2)
-    dynamoQueryEqualsValue('sourceId -> id)(
-      expectedValue = expectedSierraRecord)
+
+    eventually {
+      val futureRecord = hybridStore.getRecord[SierraTransformable](
+        expectedSierraTransformable.id)
+      whenReady(futureRecord) { record =>
+        println(s"@@AWLC The future is here!")
```
Comment (alicefuzier) [inline sierra_adapter/sierra_bib_merger/src/test/scala/uk/ac/wellcome/platform/sierra_bib_merger/SierraBibMergerFeatureTest.scala]: print

File: sierra_adapter/sierra_bib_merger/src/test/scala/uk/ac/wellcome/platform/sierra_bib_merger/SierraBibMergerFeatureTest.scala [hunk 4]
```diff
@@ -187,14 +237,19 @@ class SierraBibMergerFeatureTest
     // enough time for this update to have gone through (if it was going to).
     Thread.sleep(5000)
 
-    dynamoQueryEqualsValue('sourceId -> id)(
-      expectedValue = expectedSierraRecord)
+    eventually {
+      val futureRecord = hybridStore.getRecord[SierraTransformable](
+        expectedSierraTransformable.id)
+      whenReady(futureRecord) { record =>
+        println(s"@@AWLC The future is here!")
```
Comment (alicefuzier) [inline sierra_adapter/sierra_bib_merger/src/test/scala/uk/ac/wellcome/platform/sierra_bib_merger/SierraBibMergerFeatureTest.scala]: print

File: sierra_adapter/sierra_bib_merger/src/test/scala/uk/ac/wellcome/platform/sierra_bib_merger/SierraBibMergerFeatureTest.scala [hunk 5]
```diff
@@ -208,12 +263,23 @@ class SierraBibMergerFeatureTest
...
+    val expectedSierraTransformable =
       SierraTransformable(bibRecord = record, version = 2)
 
-    dynamoQueryEqualsValue('sourceId -> id)(
-      expectedValue = expectedSierraRecord)
+    eventually {
+      val futureRecord = hybridStore.getRecord[SierraTransformable](
+        expectedSierraTransformable.id)
+      whenReady(futureRecord) { record =>
+        println(s"@@AWLC The future is here!")
```
Comment (alicefuzier) [inline sierra_adapter/sierra_bib_merger/src/test/scala/uk/ac/wellcome/platform/sierra_bib_merger/SierraBibMergerFeatureTest.scala]: and again

File: sierra_adapter/sierra_item_merger/src/main/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterService.scala [hunk 6]
```diff
@@ -2,59 +2,79 @@ package uk.ac.wellcome.platform.sierra_item_merger.services
...
 import scala.concurrent.Future
 
 class SierraItemMergerUpdaterService @Inject()(
-  sierraTransformableDao: SierraTransformableDao,
+  versionedHybridStore: VersionedHybridStore,
   metrics: MetricsSender
 ) extends Logging {
 
-  def update(itemRecord: SierraItemRecord): Future[Unit] = {
+  lazy implicit val decoder = deriveDecoder[SierraTransformable]
```
Comment (alicefuzier) [inline sierra_adapter/sierra_item_merger/src/main/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterService.scala]: are you sure you need this?

File: sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterServiceTest.scala [hunk 7]
```diff
@@ -103,39 +117,48 @@ class SierraItemMergerUpdaterServiceTest
...
-
-    whenReady(sierraUpdaterService.update(itemRecord)) { _ =>
-      val expectedNewSierraRecord =
-        SierraTransformable(
-          sourceId = bibIdNotExisting,
-          maybeBibData = None,
-          itemData = Map(itemRecord.id -> itemRecord),
-          version = 1
+    val f2 = hybridStore.updateRecord[SierraTransformable](newRecord)
+    whenReady(Future.sequence(List(f1, f2))) { _ =>
```
Comment (alicefuzier) [inline sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterServiceTest.scala]: This doesn't enforce one happening after the other. If that's what you want you should use `f1.flatMap(_ => f2)`

Comment (kenoir) [reply sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterServiceTest.scala]: the order doesn't matter, just that they both happen before the next whenReady

File: sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterServiceTest.scala [hunk 8]
```diff
@@ -150,33 +173,36 @@ class SierraItemMergerUpdaterServiceTest
...
+    whenReady(f1) { _ =>
+      val newItemRecord = sierraItemRecord(
+        id = id,
+        updatedDate = "2014-04-04T04:04:04Z",
+        bibIds = List(bibId)
       )
 
-      dynamoQueryEqualsValue('sourceId -> bibId)(
-        expectedValue = expectedSierraRecord)
+      whenReady(sierraUpdaterService.update(newItemRecord)) { _ =>
```
Comment (alicefuzier) [inline sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/services/SierraItemMergerUpdaterServiceTest.scala]: I don't see the need for nested `whenReady` here, you can just `flatMap`

File: sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/utils/SierraItemMergerTestUtil.scala [hunk 9]
```diff
@@ -1,18 +1,41 @@
...
+      override def updateVersion(sierraTransformable: SierraTransformable,
+                                 newVersion: Int): SierraTransformable = {
+        sierraTransformable.copy(version = newVersion)
+      }
+    }
+
+  override lazy val tableName = "sierra-item-merger-feature-test-table"
+  override lazy val bucketName = "sierra-item-merger-feature-test-bucket"
+
+  lazy implicit val decoder = deriveDecoder[SierraTransformable]
```
Comment (alicefuzier) [inline sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/utils/SierraItemMergerTestUtil.scala]: I don't think you need this?

**Pull request merged**